zgw_consumers/models/lookups.py

```diff
@@ -0,0 +1,155 @@
+from typing import Optional, Tuple
+
+from django.db.models.fields.related_lookups import get_normalized_value
+from django.db.models.lookups import Exact as _Exact, In as _In
+
+from zgw_consumers.models import Service
+
+from .fields import ServiceUrlField
+
+
+def decompose_value(value: str) -> Tuple[Optional[Service], Optional[str]]:
+    service = Service.get_service(value)
+    if not service:
+        return None, None
+
+    relative_val = value[len(service.api_root) :]
+    return service, relative_val
+
+
+class ServiceUrlFieldMixin:
+    def split_lhs(self, compiler, connection) -> Tuple[str, tuple, str, tuple]:
+        target = self.lhs.target
+        alias = target.model._meta.db_table
+
+        base_lhs = target._base_field.get_col(alias)
+        relative_lhs = target._relative_field.get_col(alias)
+
+        base_lhs_sql, base_lhs_params = self.process_lhs(
+            compiler, connection, lhs=base_lhs
+        )
+        relative_lhs_sql, relative_lhs_params = self.process_lhs(
+            compiler, connection, lhs=relative_lhs
+        )
+
+        return base_lhs_sql, base_lhs_params, relative_lhs_sql, relative_lhs_params
+
+    def get_prep_lookup(self) -> list:
+        if not self.rhs_is_direct_value():
+            return super().get_prep_lookup()
+
+        target = self.lhs.target
+        alias = target.model._meta.db_table
+        base_lhs, relative_lhs = [
+            field.get_col(alias, output_field=field)
+            for field in [target._base_field, target._relative_field]
+        ]
+        value = self.rhs if self.get_db_prep_lookup_value_is_iterable else [self.rhs]
+
+        prepared_values = []
+        for rhs in value:
+            base_value, relative_value = decompose_value(rhs)
+
+            # convert model instances to int for FK fields
+            base_normalized_value = get_normalized_value(base_value, base_lhs)[0]
+            relative__normalized_value = get_normalized_value(
```

Comment:
```suggestion
            relative_normalized_value = get_normalized_value(
```

To prevent confusion (its not a relation).